src/main/java/hudson/plugins/templateproject/TemplateUtils.java

```diff
@@ -1,38 +1,52 @@
 package hudson.plugins.templateproject;
 
-import hudson.EnvVars;
+import com.google.common.annotations.VisibleForTesting;
 import hudson.Util;
 import hudson.model.AbstractBuild;
 import hudson.model.AbstractProject;
-import hudson.model.BuildListener;
-import hudson.model.TaskListener;
 import hudson.model.Hudson;
-import hudson.util.LogTaskListener;
-import static java.util.logging.Level.INFO;
-import java.util.logging.Logger;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
 
 public class TemplateUtils {
-	private static final Logger logger = Logger.getLogger("TemplateProject");
 
-	public static AbstractProject<?, ?> getProject(String projectName, AbstractBuild<?, ?> build) {
-		String pName = projectName;
+	private static TemplateUtils instance;
 
-		if (build != null) {
-			pName = TemplateUtils.getExpandedProjectName(projectName, build);
+	private final Hudson hudson;
 
-			if (Hudson.getInstance().getItemByFullName(pName) == null) {
-				logger.info("[TemplateProject] Template Project '" + pName + "' not found. Skipping.");
-			}
-		}
+	private TemplateUtils(){
+		this(Hudson.getInstance());
```

Comment:
No, the test framework I reference is no closed-source unfortunately. BTW it was just a corner-case example.

In the case of your tests I would say that you could use <code>JenkinsRule</code> instead of mocking. Such functional tests take more time, but they provide a better coverage.